### PR TITLE
Check scope for personal tokens

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: server/graphql/common/scope-check.ts [hunk 1]
```diff
@@ -116,11 +117,27 @@ export const checkRemoteUserCanRoot = (req: express.Request): void => {
 type OAuthScope = keyof typeof OAuthScopes;
 
 export const checkScope = (req: express.Request, scope: OAuthScope): boolean => {
-  return !req.userToken || req.userToken.hasScope(scope);
+  if (req.userToken) {
+    return req.userToken.hasScope(scope);
+  } else if (req.personalToken) {
+    // Personal Tokens had no scope until this date, all scopes were assumed
+    if (moment('2023-01-23') > moment(req.personalToken.createdAt)) {
```
Comment (znarf) [inline server/graphql/common/scope-check.ts:124]: 2023-01-23

Comment (znarf) [reply server/graphql/common/scope-check.ts:124]: updatedAt ?

Comment (Betree) [reply server/graphql/common/scope-check.ts:124]: Alternative: update all personal tokens created before `2023-01-23` to give them all scopes. Should be followed by a deprecation policy (users must update the tokens or they'll be removed)

**Comment by znarf** [comment]
added support for `expiresAt`

**Review by Betree — COMMENTED** [review]
Looks good to me so far

File: server/middleware/authentication.js [hunk 2]
```diff
@@ -290,11 +291,20 @@ export async function checkPersonalToken(req, res, next) {
   const token = req.get('Personal-Token') || req.query.personalToken;
 
   if (apiKey || token) {
+    const now = moment();
     const personalToken = await models.PersonalToken.findOne({
-      where: { token: apiKey || token },
+      where: {
+        token: apiKey || token,
+        [Op.or]: [{ expiresAt: { [Op.is]: null } }, { expiresAt: { [Op.lte]: now } }],
```
[the remark this answers is not in the document]
Comment (znarf) [reply server/middleware/authentication.js:298]: True.

[the remark this answers is not in the document]
Comment (znarf) [reply server/middleware/authentication.js]: PR updated

**Pull request merged**